### PR TITLE
Surface library container resolving errors

### LINKED ISSUE #3449: Failure creating native AES project with Objectify
Eclipse 2019-03.  Create New _App Engine Standard_ project with Objectify fails: no Objectify library is loaded.  The project class path doesn't resolve properly.

```
	<classpathentry kind="con" path="com.google.cloud.tools.eclipse.appengine.libraries/master-container">
		<attributes>
			<attribute name="org.eclipse.jst.component.dependency" value="/WEB-INF/lib"/>
		</attributes>
	</classpathentry>
```
and the `.../.settings/com.google.cloud.tools.eclipse.appengine.libraries/_libraries.container` looks ok:
```
[
  "objectify"
]
```

Removing the cloud libraries container and re-adding Objectify fails with a stack-trace:
```
com.google.common.base.VerifyException: expected a non-null reference
	at com.google.common.base.Verify.verify(Verify.java:124)
	at com.google.common.base.Verify.verifyNotNull(Verify.java:500)
	at com.google.common.base.Verify.verifyNotNull(Verify.java:477)
	at com.google.cloud.tools.eclipse.appengine.libraries.ui.CloudLibrariesPage.finish(CloudLibrariesPage.java:177)
	at org.eclipse.jdt.internal.ui.wizards.buildpaths.ClasspathContainerWizard.performFinish(ClasspathContainerWizard.java:95)
```

### PR DESCRIPTION
**Pull request by briandealwis**
Fixes #3449 by surfacing the error to the user.

On error:

Screen Shot 2019-06-06 at 10 50 09 AM

Expanded detail:

Screen Shot 2019-06-06 at 10 50 32 AM

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/LibraryClasspathContainerInitializer.java [hunk 1]
```diff
@@ -80,7 +81,11 @@ public void initialize(IPath containerPath, IJavaProject project) throws CoreExc
             new IClasspathContainer[] {container},
             new NullProgressMonitor());
       } else {
-        resolverService.resolveContainer(project, containerPath, new NullProgressMonitor());
+        IStatus result =
+            resolverService.resolveContainer(project, containerPath, new NullProgressMonitor());
+        if (!result.isOK()) {
+          throw new CoreException(result);
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/LibraryClasspathContainerInitializer.java]: the default message isn't very helpful. Could we provide a better custom message here?

**Comment by chanseokoh** [comment]
Looks like integration tests are failing much more frequently. Seems very unusual. Cloud it be because of #3448?

**Comment by briandealwis** [comment]
I think those test failures are good ol' deadlocks: our LibraryClasspathContainerInitializer is very heavy-weight and acquires synchronization locks but then blocks when attempting to obtain workspace locks.

```
Eclipse Locks:
|  :: 
|  R/,
|  Worker-6: Refreshing workspace :  -1,
|  Worker-15: Building workspace :  1,
|  Worker-9: Compacting resource model :  -1,
```

(The above means that Worker-15 holds the lock on the workspace root `R/`, and Worker-6 and Worker-9 are waiting for that lock.)
```
...
| "Worker-6: Refreshing workspace" [53] WAITING on java.lang.Object@6c3c7be0
|     at java.lang.Object.wait(Native Method)
|     - waiting on java.lang.Object@6c3c7be0
|     at java.lang.Object.wait(Object.java:502)
|     at org.eclipse.core.internal.jobs.ThreadJob.waitForRun(ThreadJob.java:313)
|     at org.eclipse.core.internal.jobs.ThreadJob.joinRun(ThreadJob.java:202)
|     at org.eclipse.core.internal.jobs.ImplicitJobs.begin(ImplicitJobs.java:92)
|     at org.eclipse.core.internal.jobs.JobManager.beginRule(JobManager.java:308)
|     at com.google.cloud.tools.eclipse.util.MavenUtils.runOperation(MavenUtils.java:113)
|     at com.google.cloud.tools.eclipse.util.DependencyResolver.getTransitiveDependencies(DependencyResolver.java:60)
|     at com.google.cloud.tools.eclipse.appengine.libraries.model.LibraryFactory.loadTransitiveDependencies(LibraryFactory.java:120)
|     at com.google.cloud.tools.eclipse.appengine.libraries.model.Library.resolveDependencies(Library.java:211)
|     - locked com.google.cloud.tools.eclipse.appengine.libraries.model.Library@66c41090
|     at com.google.cloud.tools.eclipse.appengine.libraries.model.Library.getAllDependencies(Library.java:117)
|     - locked com.google.cloud.tools.eclipse.appengine.libraries.model.Library@66c41090
|     at com.google.cloud.tools.eclipse.appengine.libraries.repository.LibraryClasspathContainerResolverService.resolveLibraryFiles(LibraryClasspathContainerResolverService.java:191)
|     at com.google.cloud.tools.eclipse.appengine.libraries.repository.LibraryClasspathContainerResolverService.resolveContainer(LibraryClasspathContainerResolverService.java:164)
|     at com.google.cloud.tools.eclipse.appengine.libraries.LibraryClasspathContainerInitializer.initialize(LibraryClasspathContainerInitializer.java:85)
|     at org.eclipse.jdt.internal.core.JavaModelManager.initializeContainer(JavaModelManager.java:3184)
...
| "Worker-15: Building workspace" [183] BLOCKED on com.google.cloud.tools.eclipse.appengine.libraries.model.Library@66c41090 owned by 'Worker-6: Refreshing workspace [id:53]
|     at com.google.cloud.tools.eclipse.appengine.libraries.model.Library.getAllDependencies(Library.java)
|     - blocked on com.google.cloud.tools.eclipse.appengine.libraries.model.Library@66c41090
|     at com.google.cloud.tools.eclipse.appengine.libraries.repository.LibraryClasspathContainerResolverService.resolveLibraryFiles(LibraryClasspathContainerResolverService.java:191)
|     at com.google.cloud.tools.eclipse.appengine.libraries.repository.LibraryClasspathContainerResolverService.resolveContainer(LibraryClasspathContainerResolverService.java:164)
|     at com.google.cloud.tools.eclipse.appengine.libraries.LibraryClasspathContainerInitializer.initialize(LibraryClasspathContainerInitializer.java:85)
|     at org.eclipse.jdt.internal.core.JavaModelManager.initializeContainer(JavaModelManager.java:3184)
|     at org.eclipse.jdt.internal.core.JavaModelManager$10.run(JavaModelManager.java:3076)

```

**Comment by briandealwis** [comment]
Maybe it's due to interference from this new exception.

**Comment by briandealwis** [comment]
Seems this change exacerbates the situation as the [master build](https://travis-ci.org/GoogleCloudPlatform/google-cloud-eclipse/builds/542070239) was fine.  Hmm maybe we should just call `StatusUtil.setErrorStatus()`

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/LibraryClasspathContainerInitializer.java (hunk 1 above)
Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/LibraryClasspathContainerInitializer.java]: I've improved the message slightly; suggestions welcome.

**Comment by briandealwis** [comment]
Let's see if using `StatusUtil.setErrorStatus()` improved things.  I changed how `StatusUtil.setErrorStatus()` reports its text as adding the exception message from resulted in a big jumble of text that was hard to parse.

**Comment by briandealwis** [comment]
Abandoning this fix.
